tensorflow_lite_support/cc/task/processor/image_postprocessor.cc

```diff
@@ -0,0 +1,237 @@
+using ::tflite::support::StatusOr;
+using ::tflite::support::TfLiteSupportStatus;
+
+StatusOr<absl::optional<vision::NormalizationOptions>>
+GetNormalizationOptionsIfAny(const TensorMetadata& tensor_metadata) {
+  ASSIGN_OR_RETURN(
+      const tflite::ProcessUnit* normalization_process_unit,
+      ModelMetadataExtractor::FindFirstProcessUnit(
+          tensor_metadata, tflite::ProcessUnitOptions_NormalizationOptions));
+  if (normalization_process_unit == nullptr) {
+    return {absl::nullopt};
+  }
+  const tflite::NormalizationOptions* tf_normalization_options =
+      normalization_process_unit->options_as_NormalizationOptions();
+  const auto mean_values = tf_normalization_options->mean();
+  const auto std_values = tf_normalization_options->std();
+  if (mean_values->size() != std_values->size()) {
+    return CreateStatusWithPayload(
+        StatusCode::kInvalidArgument,
+        absl::StrCat("NormalizationOptions: expected mean and std of same "
+                     "dimension, got ",
+                     mean_values->size(), " and ", std_values->size(), "."),
+        TfLiteSupportStatus::kMetadataInvalidProcessUnitsError);
+  }
+  absl::optional<vision::NormalizationOptions> normalization_options;
+  if (mean_values->size() == 1) {
+    normalization_options = vision::NormalizationOptions{
+        .mean_values = {mean_values->Get(0), mean_values->Get(0),
+                        mean_values->Get(0)},
+        .std_values = {std_values->Get(0), std_values->Get(0),
+                       std_values->Get(0)},
+        .num_values = 1};
+  } else if (mean_values->size() == 3) {
+    normalization_options = vision::NormalizationOptions{
+        .mean_values = {mean_values->Get(0), mean_values->Get(1),
+                        mean_values->Get(2)},
+        .std_values = {std_values->Get(0), std_values->Get(1),
+                       std_values->Get(2)},
+        .num_values = 3};
+  } else {
+    return CreateStatusWithPayload(
+        StatusCode::kInvalidArgument,
+        absl::StrCat("NormalizationOptions: only 1 or 3 mean and std "
+                     "values are supported, got ",
+                     mean_values->size(), "."),
+        TfLiteSupportStatus::kMetadataInvalidProcessUnitsError);
+  }
+  return normalization_options;
+}
+}  // namespace
+
+/* static */
+tflite::support::StatusOr<std::unique_ptr<ImagePostprocessor>>
+ImagePostprocessor::Create(core::TfLiteEngine* engine,
+                           const std::initializer_list<int> output_indices,
+                           const std::initializer_list<int> input_indices) {
+  ASSIGN_OR_RETURN(auto processor,
+                   Processor::Create<ImagePostprocessor>(
+                       /* num_expected_tensors = */ 1, engine, output_indices,
+                       /* requires_metadata = */ false));
+
+  RETURN_IF_ERROR(processor->Init(input_indices));
+  return processor;
+}
+
+absl::Status ImagePostprocessor::Init(const std::vector<int> &input_indices) {
```

Comment:
Still, this Init function is largely the same as `ImageTensorSpecs::BuildInputImageTensorSpecs`. We should update `BuildInputImageTensorSpecs` and reuse the logic between `ImagePostprocessor::Init` and `ImagePreprocessor::Init` as much as possible.

Reply:
Understood, instead of passing the tensor, we pass a boolean switch to handle cases like [these](https://github.com/tensorflow/tflite-support/blob/f5dadc83b36d863700f080ca45877f8b3b3d7f47/tensorflow_lite_support/cc/task/vision/utils/image_tensor_specs.cc#L140-L141) & rename to ~BuildInputImageTensorSpecs~ =>  BuildImageTensorSpecs
```c++
StatusOr<ImageTensorSpecs> BuildImageTensorSpecs(
    const TfLiteEngine::Interpreter& interpreter,
    const tflite::metadata::ModelMetadataExtractor& metadata_extractor,
    const bool is_input) { // Notice the boolean
```

similarly for ~GetInputTensorMetadataIfAny~ => GetTensorMetadataIfAny [here](https://github.com/tensorflow/tflite-support/blob/f5dadc83b36d863700f080ca45877f8b3b3d7f47/tensorflow_lite_support/cc/task/vision/utils/image_tensor_specs.cc#L42).

Such that we could switch getting InputCount vs OutputCount, InputTensor access vs OutputTensor access easily. How's the plan?

Reply:
Okay, we decided against it.